### DISCUSSION TITLE
Ask accessibility for barriers in a quest?

### DISCUSSION THREAD

**Discussion by mcliquid** [streetcomplete/StreetComplete#5151] [Ideas]
Before I open a new issue for a quest idea, and I've already seen that the topic has already been discussed in a slightly different way, I'd like to start with a discussion first to get your opinion on it.

I keep coming across different barriers (mostly gates) when cycling or hiking on official ways. Just this weekend it was a door in a fence. The door was open and there was not even a lock installed in it, so also not lockable. From afar, however, I already had the fear that I can not get through here. But it was open:
![image](https://github.com/streetcomplete/StreetComplete/assets/3351668/1aaeecf1-9808-4fae-a9cb-6d6b4d8530fe)

Another situation was in South Tyrol. According to the signage, it is forbidden for all vehicles. The path is a connection between the main road and the bike path that goes along further back (behind the small sheds). The large gate is locked, pedestrians have to squeeze through on the right. You also get your bike through, but it's not wanted so a `vehicle=no` bzw. `vehicle=permit` would be appropriate here:
![image](https://github.com/streetcomplete/StreetComplete/assets/3351668/e6db84d6-b01a-4f0f-a708-bce935f76250)

A similar situation in Austria, where pedestrians are allowed but cyclists and other vehicles are prohibited:
![image](https://github.com/streetcomplete/StreetComplete/assets/3351668/170c9c62-9c87-4681-8d76-40d3a12206a1)

I would be interested in capturing whether pedestrians or bicyclists come through there. This is especially the case with `barrier=gate`.
The quest would look similar like "[Prohibited for pedestrians?](app/src/main/java/de/westnordost/streetcomplete/quests/foot/AddProhibitedForPedestrians.kt)".
There would probably be two quests:

- Can pedestrians go through here?
- Can cyclists ride through here?

What do you folks think about these quests?

Similiar issues:
#1141 
#2040

**Comment by mnalis** [comment]
I seem to recall discussion on community.osm.org (which I am, typically, unable to find now) which was discussing what tagging like `barrier=swing_gate` + `locked=yes` + `foot=yes` node on `highway=service` way even means (is it '"only pedestrian can pass, after they unlock it"' or '"all vehicles can pass after they unlock, but pedestrians may pass it even without unlocking"') with additional discussion fork about _"access tags are only about legal access rights, not about a physical possibility"_. 

Which is also confusing in your first picture example, are you talking whether you **can** pass through, i.e. about physical possibility ( `width=*`/`wheelchair=*`/...) or whether you **may** pass through (i.e. legality of such act).

For third picture, I'd leave a SC Picture Note, and later map it as two separate ways, each with its own barrier and access rights.

Anyway, I think such **quest would probably fail few of the SC guidelines**. It might not be trivial for SC mapper (especially if they're not well versed in that country traffic laws - which may not be their home country either, e.g. tourists) to know what it allowed, and what is not. Also many might not be explicitly signed. And quite a few seem to be of _"it's complicated"_ variety.

In SC, I'd probably settle for _"long press to leave note"_, but perhaps "access overlay" might be more useful to add such restrictions to nodes/ways instead of a quest (as the users would then only invoke them when the situation was clear and easy to answer), although there days I'd probably switch to EveryDoor (or OsmAnd as I usually have it running and recording track already, and not needing to download things) for editing such cases. 

On the other hand, the Overlays are cumbersome in SC, so I understand why quest would be much preferred - the initial idea was _"I only care about one thing today, and that is all that I will do"_ (which might be the case in some specific use cases, but IMHO not in majority of them where users end up constantly juggling over dozes of overlays every few seconds. Perhaps [quick overlay button](https://github.com/streetcomplete/StreetComplete/pull/5109) may alleviate that at least somewhat)

**Reply by mnalis** [reply]
> perhaps "access overlay" might be more useful to add such restrictions to nodes/ways instead of a quest (as the users would then only invoke them when the situation was clear and easy to answer),

FYI, for those interested, in the meantime [SCEE](https://wiki.openstreetmap.org/wiki/SCEE) (StreetComplete _"Expert edition"_ fork) got additional option to set access on any relevant way (to be used with any relevant quests or overlay) to `Add access...`, which offers even wider configurability (as well as more widespread and less ambiguous support than tagging access on barrier nodes), e.g.:

![small_Screenshot_20250426_121618_SCEE](https://github.com/user-attachments/assets/598762a2-0472-4b35-b4f1-0f4ac995282d) ![small_Screenshot_20250426_121715_SCEE](https://github.com/user-attachments/assets/5e820309-9c86-43b0-8c7e-d6bcf89b9a89)

**Comment by matkoniecz** [comment]
This seems promising - though I am curious is it possible to reliably detect ones where it is worth asking.
